dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/ProgramRuleVariableObjectBundleHook.java

```diff
@@ -94,18 +97,47 @@
     private void validateUniqueProgramRuleName( ObjectBundle bundle,
         ProgramRuleVariable programRuleVariable, Consumer<ErrorReport> addReports )
     {
-        Query<ProgramRuleVariable> query = getProgramRuleVariableQuery( programRuleVariable );
+        List<ProgramRuleVariable> prvWithSameNameAndSameProgram = getProgramRuleVariableQuery( programRuleVariable )
+            .getResultList();
```

Comment:
I think this approach is too verbose, in my opinion we should simply try to do a query also using prv UID in the query.
After that if it is a create strategy and we get a record, we have an error. We also have an error if we don't get a result and it is an update.
I think this way is much cleaner, less code and maintenable

Reply:
you're right, I'll review it

Reply:
better thinking, we should
- fail when it's an insert and a PRV with same name/program already exists
- fail when it's an update and a PRV with same name/program/uid already exists
so I think it's not possible to verify the first check adding UID in the query filter